File: sdk/typescript/src/extensions.ts
/**
 * TypeScript SDK Client for Extension Framework.
 */

export class ExtensionsClient {
  private baseUrl: string;

  constructor(baseUrl: string = "http://localhost:8000") {
    this.baseUrl = baseUrl.replace(/\/$/, "");
  }

  async listExtensions(): Promise<Record<string, any>> {
    const res = await fetch(`${this.baseUrl}/v1/extensions`);
    if (!res.ok) throw new Error(`Fetch failed: ${res.statusText}`);
    return res.json();
  }
}

File: sdk/go/extensions.go
package llmengine

import (
	"encoding/json"
	"fmt"
	"net/http"
)

type ExtensionsClient struct {
	BaseURL    string
	HTTPClient *http.Client
}

func NewExtensionsClient(baseURL string) *ExtensionsClient {
	return &ExtensionsClient{
		BaseURL:    baseURL,
		HTTPClient: &http.Client{},
	}
}

func (c *ExtensionsClient) ListExtensions() (map[string]interface{}, error) {
	resp, err := c.HTTPClient.Get(fmt.Sprintf("%s/v1/extensions", c.BaseURL))
	if err != nil {
		return nil, err
	}
	defer resp.Body.Close()

	var result map[string]interface{}
	if err := json.NewDecoder(resp.Body).Decode(&result); err != nil {
		return nil, err
	}
	return result, nil
}
